controllers/configuration_controller.go

```diff
@@ -89,16 +95,33 @@ func (r *ConfigurationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 	klog.InfoS("reconciling Terraform Configuration...", "NamespacedName", req.NamespacedName)
 	applyJobName := configurationName + "-" + string(TerraformApply)
 
-	// Terraform destroy
 	if err := r.Get(ctx, req.NamespacedName, &configuration); err != nil {
-		klog.InfoS("performing Terraform Destroy", "Namespace", req.Namespace, "Name", req.Name)
 		if kerrors.IsNotFound(err) {
+			klog.ErrorS(err, "unable to fetch Configuration", "NamespacedName", req.NamespacedName)
+			err = nil
```

Comment:
If Configuration is deleted, all its child resources `job` and `pod` should be deleted. Right? Now nothing will do, and `job` and `pod` will be orphaned.

Reply:
job and pod have both been deleted.

Reply:
How those children are deleted at the moment Configuration is deleted?

Reply:
I mean when I did `kubectl delete Configuration xxx`, the Configuration object will become nil when retrieved by Get method.

Reply:
Those children are deleted after the destroyJob is completed and before the finalizer was removed.
If the destroyJob is not completed,  controller will job this configuration in the request queue.
Wait 3 second.

Reply:
Because add the finalizer, controller can not delete configuration directly.  
Configuration will be deleted, until the finalizer is remove.

Reply:
https://book.kubebuilder.io/reference/using-finalizers.html